### ISSUE TITLE
Micro stuttering after opening/closing the OSD menu

### ISSUE THREAD

**Issue by diedel** [openMSX/openMSX#1237]
I've noticed that openMSX micro stutters after opening and closing the OSD Menu while playing a game. If I pause/unpause the emulation the micro stuttering is gone.

Testing environment:
- Windows 10 Home
- Nvidia GTX 1080 with latest Nvidia drivers
- openMSX 0.15.0 and several FiXato builds after that version, both 32 and 64 bits
- C-BIOS MSX2+ NTSC and Philips VG 8020 PAL
- River Raid GoodMSX and Road Fighter GoodMSX

**Comment by MBilderbeek** [comment]
This could just be the OSD menu taking a little while to get drawn or removed.

Do you consider this to be an important issue?

**Comment by m9710797** [comment]
@MBilderbeek: I think diedel means permanent stuttering after opening/closing the menu, not only during those actions?
@diedel: correct?

No idea what's going on though.

**Comment by diedel** [comment]
Yes, I mean the microstuttering is permanent after opening the menu and closing it.

Meanwhile I did more tests and the microstuttering appears not only after opening/closing the menu. Also it appears frequently just after loading a game via command line or from the console (F10).
In all cases , pausing the emulator with F9 solves the problem,

Can you reproduce the problem?
Perhaps it's worth noting that I have a Gsync monitor. Also that this only happens with openMSX emulator.

**Comment by MBilderbeek** [comment]
We are both on Linux, so we cannot try on Windows. Can you confirm you used the latest development builds as well?
Did I understand correctly that pausing and unpausing solves the problem? (F9 isn't pause... so I'm confused.)

**Comment by diedel** [comment]
Sorry, I mean PAUSE key.

**Comment by MBilderbeek** [comment]
I did experience a problem similar to this, see #945 - it also includes stutters.

Can you see whether the framerate is also affected? When it happens, does it help to change the samples setting a bit (see #945)?

**Comment by diedel** [comment]
More tests.... :)

If during fullscreen I press F12 to go windowed and press it again to turn fullscreen, the microstuttering appears ALWAYS.

**Comment by diedel** [comment]
> 
> 
> I did experience a problem similar to this, see #945 - it also includes stutters.
> 
> Can you see whether the framerate is also affected?

If I check the fps OSD from openMSX the framerate moves slowly between 58 and 61 fps. Not very different from the normal state.
But if I check the OSD FPS of the monitor the framerate moves faster and seems more unstable than in the normal state.

**Comment by diedel** [comment]
> 
> 
> When it happens, does it help to change the samples setting a bit (see #945)?

Yes, setting the samples from 2048 to 2049 removes the microstuttering.

Disabling the sound with mute (F11) or setting sound_driver to null, even prevents the microstuttering from appearing.

Seems the audio and video lost sync...

**Comment by MBilderbeek** [comment]
The idea was also to unmute and set the sound_driver back to the original value.

**Comment by diedel** [comment]
In openMSX 0.16.0 it seems the permanent micro stuttering due to open/close the OSD or enable/disable fullscreen is fixed.

But that micro stuttering still appears occasionally during normal playing, after some time. Do I open another issue (with a more explanatory title) or continue with this one?

**Comment by MBilderbeek** [comment]
@diedel Let's open a new issue and close this one. And then please give all details and what influences it, etc.
